torch/cuda/amp/autocast_mode.py

```diff
@@ -112,31 +122,37 @@ def forward(self, input):
     Args:
         enabled(bool, optional, default=True):  Whether autocasting should be enabled in the region.
     """
-    def __init__(self, enabled=True):
-        if enabled and not torch.cuda.is_available():
+
+    def __init__(self, enabled: bool = True):
+        if torch._jit_internal.is_scripting():
+            self._enabled = enabled
+        elif enabled and not torch.cuda.is_available():
             warnings.warn("torch.cuda.amp.autocast only affects CUDA ops, but CUDA is not available.  Disabling.")
             self._enabled = False
         else:
             self._enabled = enabled
 
     def __enter__(self):
+        if torch._jit_internal.is_scripting():
+            return self
```

Comment:
Why do you bother returning self here? The other code path doesn't return anything (so None).

Reply:
If my memory serves me well, I think it was a workaround for a small TorchScript issue that got resolved in the meantime - so I'll remove it.

Reply:
@mcarilli - I remember now the reason for the change: we need the return value in order to support `with x as y`
(https://docs.python.org/3/reference/datamodel.html#object.__enter__)

ex. 

```python
with autocast() as a:
  ...
```